stan/math/rev/fun/multiply_log.hpp

```diff
@@ -105,6 +105,265 @@ inline var multiply_log(double a, const var& b) {
   return var(new internal::multiply_log_dv_vari(a, b.vi_));
 }
 
+/**
+ * Return the elementwise product `a * log(b)`.
+ *
+ * Both `T1` and `T2` are matrices, and one of `T1` or `T2` must be a
+ * `var_value`
+ *
+ * @tparam T1 Type of first argument
+ * @tparam T2 Type of second argument
+ * @param a First argument
+ * @param b Second argument
+ * @return elementwise product of `a` and `log(b)`
+ */
+template <typename T1, typename T2, require_all_matrix_t<T1, T2>* = nullptr,
+          require_any_var_matrix_t<T1, T2>* = nullptr>
+inline auto multiply_log(const T1& a, const T2& b) {
+  check_matching_dims("multiply_log", "a", a, "b", b);
+  if (!is_constant<T1>::value && !is_constant<T2>::value) {
+    arena_t<promote_scalar_t<var, T1>> arena_a = a;
+    arena_t<promote_scalar_t<var, T2>> arena_b = b;
+
+    return make_callback_var(
+        multiply_log(arena_a.val(), arena_b.val()),
+        [arena_a, arena_b](const auto& res) mutable {
+          for (Eigen::Index j = 0; j < res.adj().cols(); ++j) {
+            for (Eigen::Index i = 0; i < res.adj().rows(); ++i) {
+              if (unlikely(is_any_nan(arena_a.val().coeff(i, j),
+                                      arena_b.val().coeff(i, j)))) {
+                arena_a.adj().coeffRef(i, j) = NOT_A_NUMBER;
+                arena_b.adj().coeffRef(i, j) = NOT_A_NUMBER;
+              } else {
+                arena_a.adj().coeffRef(i, j)
+                    += res.adj().coeff(i, j) * log(arena_b.val().coeff(i, j));
+                if (arena_b.val().coeff(i, j) == 0.0
+                    && arena_a.val().coeff(i, j) == 0) {
+                  arena_b.adj().coeffRef(i, j) += res.adj().coeff(i, j) * INFTY;
+                } else {
+                  arena_b.adj().coeffRef(i, j) += res.adj().coeff(i, j)
+                                                  * arena_a.val().coeff(i, j)
+                                                  / arena_b.val().coeff(i, j);
+                }
+              }
+            }
+          }
+        });
+  } else if (!is_constant<T1>::value) {
+    arena_t<promote_scalar_t<var, T1>> arena_a = a;
+    auto arena_b = to_arena(value_of(b));
```

Comment:
I'd rather just have these do the same pattern
```suggestion
    arena_t<promote_scalar_t<var, T1>> arena_a = a;
    arena_t<promote_scalar_t<double, T2>> arena_b = value_of(b);
```